src/lib/cfa/sudoers.rb

```diff
@@ -0,0 +1,237 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yast2/target_file"
+require "cfa/base_model"
+require "cfa/augeas_parser"
+require "cfa/matcher"
+require "cfa/placer"
+
+module CFA
+  # Model to handle users specifications in the /etc/sudoers file
+  #
+  # @example Loading specifications
+  #   file = Sudoers.new
+  #   file.load
+  #   file.current_user_specs
+  #
+  # @example Shortcut loading specificitions
+  #   Sudoers.load.users_specs
+  #
+  # @example Writing new specifications
+  #   file = Sudores.new
+  #   file.load
+  #   file.clean_users
+  #   file.add_user({ "user" => "yast", "host" => "ALL", "commands" => ["/sbin/yast2"] })
+  #   file.add_user({ "user" => "foo", "host" => "ALL", "commands" => ["ALL"], "tag" => "NOPASSWD" })
+  #   file.save
+  class Sudoers < BaseModel
+    # @return [AgueasParser] the Augeas parser for sudoers file
+    PARSER = AugeasParser.new("sudoers.lns")
+    private_constant :PARSER
+
+    # @return [String] default path to the sudoers file
+    DEFAULT_PATH = "/etc/sudoers".freeze
+    private_constant :DEFAULT_PATH
+
+    # @return [Matcher] matcher for the users collection
+    USERS_MATCHER = Matcher.new(collection: "spec")
+    private_constant :USERS_MATCHER
+
+    class << self
+      # Instantiates and loads a file
+      #
+      # This method is basically a shortcut to instantiate and load the content in just one call.
+      #
+      # @param file_handler [#read,#write] something able to read/write a string (like File)
+      # @return [Sudoers] File with the already loaded content
+      def load(file_path: DEFAULT_PATH, file_handler: Yast::TargetFile)
```

Comment:
OK, so few points I have to this method:

1. name is confusing. I prefer something like `read` so it avoid confusion between `.load` and `#load` that has different params and return.
2. in return documentation, you do not return `File` but `Model` and for sure you should document that it is always new instance so `Sudoers.load !== Sudoers.load`
3. I do not like redefining file_handler in this method to use always Yast::TargetFile instead of nil. As it would make possible reuse of that CFA model outside of yast harder.